### PR TITLE
Add explanation for what deep links are

### PR DESCRIPTION
**Pull request by mit-mit**
_Description of what this PR is changing or adding, and why:_
Adds a small explanation for what "deep links" are before getting into the details.

_Issues fixed by this PR (if any):_ none.

_PRs or commits this PR depends on (if any):_ none.

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/content/ui/navigation/deep-linking.md [hunk 1]
```diff
@@ -2,8 +2,12 @@
 title: Deep linking
 description: Navigate to routes when the app receives a new URL.
 ---
+Deep links are links that not only open an app, but also take the
+user to a specific location "deep" inside the app. For example,
+a deep link from an advertizement for a pair of sneakers might open
+a shopping app and display the product page for those particular shows.
```
Comment (sfshaza2) [inline src/content/ui/navigation/deep-linking.md:8]:
```suggestion
a shopping app and display the product page for those particular shoes.
```

**Review by sfshaza2 — APPROVED** [review]
LGTM minus one typo

File: src/content/ui/navigation/deep-linking.md [hunk 2]
```diff
@@ -2,8 +2,12 @@
 title: Deep linking
 description: Navigate to routes when the app receives a new URL.
 ---
+Deep links are links that not only open an app, but also take the
+user to a specific location "deep" inside the app. For example,
+a deep link from an advertizement for a pair of sneakers might open
```
Comment (atsansone) [inline src/content/ui/navigation/deep-linking.md:7]: **todo:** Spelling mistake.
```suggestion
a deep link from an advertisement for a pair of sneakers might open
```

**Review by atsansone — APPROVED** [review]
@mit-mit : One spelling mistake, otherwise LGTM.

**Comment by sfshaza2** [comment]
Yes, our writing guide specifies American spelling.

**Pull request merged**